Sources/TuistCore/Models/DestinationPlatform.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+import XcodeGraph
+
+/// Represents a platform either for a simulator or a device
+public enum DestinationPlatform: Codable, Equatable {
```

Comment:
Nitpick: What about naming this one `DestinationType` instead of `DestinationPlatform`, since we are already using [`Platform`](https://github.com/tuist/tuist/blob/main/Sources/ProjectDescription/Platform.swift) elsewhere with the semantics of OS?